ibis/backends/snowflake/__init__.py

```diff
@@ -499,3 +501,14 @@ def _get_temp_view_definition(
         self, name: str, definition: sa.sql.compiler.Compiled
     ) -> str:
         yield f"CREATE OR REPLACE TEMPORARY VIEW {name} AS {definition}"
+
+
+@compiles(sa.sql.Join, "snowflake")
+def compile_join(element, compiler, **kw):
+    result = compiler.visit_join(element, **kw)
+
+    # snowflake doesn't support lateral joins with ON clauses as of
+    # https://docs.snowflake.com/en/release-notes/bcr-bundles/2023_04/bcr-1057
+    if element.right._is_lateral:
+        return re.sub(r"^(.*)\s+ON true$", r"\1", result, flags=re.IGNORECASE)
```

Comment:
is there ever a scenario in which (in code we generate) that there will be multiple whitespace characters between `ON` and `true`?

Reply:
Not that I am aware of. Actually, it shouldn't matter since I'm using `.*` which matches whitespace.